lib/system/io.nim

```diff
@@ -792,20 +797,24 @@ when declared(stdout):
         proc flockfile(f: File) {.importc, nodecl.}
         proc funlockfile(f: File) {.importc, nodecl.}
         flockfile(stdout)
+        defer: funlockfile(stdout)
       when defined(windows) and compileOption("threads"):
         acquireSys echoLock
+        defer: releaseSys echoLock
       for s in args:
         when defined(windows):
-          writeWindows(stdout, s)
+          writeWindows(stdout, s, doRaise = echoDoRaise)
         else:
-          discard c_fwrite(s.cstring, cast[csize_t](s.len), 1, stdout)
+          if c_fwrite(s.cstring, cast[csize_t](s.len), 1, stdout) != s.len:
```

Comment:
this seems unfortunately subject to the classic EINTR bug, see https://github.com/angrave/SystemProgramming/wiki/POSIX%2C-Part-1%3A-Error-handling#what-is-eintr-what-does-it-mean-for-sem_wait-read-write and how I fixed it here https://github.com/nim-lang/Nim/pull/13232 via a while loop which is IIUC the correct way to handle this

the question is whether to fix this in this PR or merge this PR as is and fix it in future PR; is there an easy repro?

Reply:
I think do it as a separate PR, abstracting out a common template in the process (as mentioned in your fix for the linked issue).